### PR TITLE
Add Draggable.onDragStarted

### LINKED ISSUE #6084: No way to know when dragging starts on a Draggable or LongPressDraggable (to trigger an animation).

### PR DESCRIPTION
**Pull request by abarth**
Fixes #6084

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 1]
```diff
@@ -242,6 +248,9 @@ class _DraggableState<T> extends State<Draggable<T>> {
         });
       }
     );
+    if (config.onDragStarted != null)
+      config.onDragStarted();
+    return avatar;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/drag_target.dart]: why do you build it before calling the callback?

Comment (abarth) [inline packages/flutter/lib/src/widgets/drag_target.dart]: paranoia about being re-entered during the callback.

Comment (Hixie) [inline packages/flutter/lib/src/widgets/drag_target.dart]: yeah ok

**Comment by Hixie** [comment]
LGTM. I think we'll eventually want to expose some version of the avatar in this callback, but let's wait for use cases.

**Pull request merged**